### PR TITLE
Update function signatures and add jsdocs

### PR DESCRIPTION
**Pull request by wemeetagain**
- Update function signatures to conform to lodestar consumer expectations (mostly have exported functions have Buffer input/outputs)
- Update function/variable names to camelCase
- Add preliminary jsdocs to exported functions
- Update hashToG2 to use mcl.Fp2#mapToG2
- Remove getHexStr (since we're outputing Buffers we can do output.toString('hex') as needed)
- Update tests to pass

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,135 +1,170 @@
...
+	return mcl.pairing(pubkeyG1, hashToG2(messageHash, domain)).isEqual(mcl.pairing(g1(), signatureG2))
 }
 
-// pubkey:bls.PublicKey()
-// message:string
-// signature:bls.Signature()
-// domain:uint
-const verify = (pub, message, sig, domain) => {
-	if (!(pub instanceof mcl.G1)) throw new Error('verify:wrong pubkey type')
-	if (!(sig instanceof mcl.G2)) throw new Error('verify:wrong signature type')
```
Comment (noot) [inline index.js:105]: should we keep these checks?

File: index.js [hunk 2]
```diff
@@ -1,135 +1,170 @@
...
-// pubkeys:list[mcl.G1]
-// messages:list[string]
-// signature:mcl.G2
-// domain:int
-const verify_multiple = (pubkeys, messages, signature, domain) => {
-	pubkeys.map((pub) => {
-        	if (!(pub instanceof mcl.G1)) throw new Error('verify_multiple:wrong pubkey type')
-	})
-	if (!(signature instanceof mcl.G2)) throw new Error('verify_multiple:wrong signature type')
-	if (pubkeys.length !== messages.length) throw new Error('verify_multiple:number of pubkeys does not equal number of messages')
```
Comment (noot) [inline index.js:120]: same with these

File: test/bls_test.js [hunk 3]
```diff
@@ -1,146 +1,110 @@
...
-        assert(s.getStr(16) === secret)
-
-        let P = bls.gen_public(s)
-        let Pstr = bls.getHexStr(P)
-
-        assert(public === Pstr, "incorrect deserialization")
-        assert(Pstr.length === 98, "wrong length")
-	})
+		const secret = '263dbd792f5b1be47ed85f8938c0f29586af0d3ac7b977f21c278fe1462040e3'
+		const pub = '9af7a7e5f618067c064d65771b0ce1d60fe2a6b9ff8d364f64e2e573a8c4ea2204ea0d4fe7f0897991bbd9ecb0d19104'
```
Comment (noot) [inline test/bls_test.js]: why is the pubkey changed here? this test was from the test vectors https://github.com/ethereum/eth2.0-tests/blob/master/bls/test_bls.yml same with why the other tests are gone

File: test/bls_test.js [hunk 4]
```diff
@@ -1,146 +1,110 @@
...
 	})
 
 	it("should hash message to G2", () => {
-		let domain = 0
-		let message = '0x6d657373616765'
-		console.log(bls.hash_to_G2(message, domain))
+		const domain = 0
+		const message = keccak256(Buffer.from('6d657373616765', 'hex'))
+		assert(bls.hashToG2(message, domain))
 	})
```
Comment (noot) [inline test/bls_test.js:21]: this is fine for now but we eventually need to make sure it conforms to the spec test vectors

**Review by noot — COMMENTED** [review]
looks great, thanks for the work Cayman :) just a few comments about the tests but the code itself looks fine

File: index.js (hunk 1 above)
Comment (wemeetagain) [reply index.js:105]: Changed the verify function to take Buffers as input.
Then we create the G1 and G2 objects inside the verify function.

Perhaps we can do a length check on the input Buffers? make sure they are 48 and 96 bytes long?

File: index.js (hunk 2 above)
Comment (wemeetagain) [reply index.js:120]: see above comment, I'm thinking we can do a similar thing by checking the length of the input Buffers.

File: test/bls_test.js (hunk 3 above)
Comment (wemeetagain) [reply test/bls_test.js]: I didn't know where they came from and thought they were not needed :disappointed: 
I'll add them back in.

File: test/bls_test.js (hunk 4 above)
Comment (wemeetagain) [reply test/bls_test.js:21]: Yeah, I was thinking that the code is still in a proto-working stage, this PR is mainly to support the input/output types that lodestar is expecting, so we can start using this library (even if the logic is not quite solid yet).

File: test/bls_test.js (hunk 3 above)
Comment (noot) [reply test/bls_test.js]: no worries! I probably should have made a comment lol.

File: test/bls_test.js (hunk 4 above)
Comment (noot) [reply test/bls_test.js:21]: yep, that is reasonable.

File: index.js (hunk 1 above)
Comment (noot) [reply index.js:105]: yeah, that would be good in case of invalid input.

**Comment by wemeetagain** [comment]
Updated w/ sk->Pk test cases from the spec tests and input length assertions

**Pull request merged**